Feature/bar chart

totzk9 opened the pull request:
# 📲 What #

* Implemented BarChart of type Vertical Stacked and Normal.

# 🤔 Why #

* New feature.

# 🛠 How #
 
* In BarChart
  - In drawData(...), draw a different path whose draw style is fill if type is BarChartType.VERTICAL or BarChartType.VERTICAL_STACKED.
  - Added composable preview for BarChartType.VERTICAL.

# ✅ Acceptance criteria #

- [x] Composable preview added
- [x] Added UI Tests

# ⏰ TODO #

* None

# 🔗 References #
<img width="309" alt="Screen Shot 2021-04-26 at 3 02 18 PM" src="https://user-images.githubusercontent.com/24382050/116042144-a9e0eb00-a6a0-11eb-92be-a366f779cb63.png">
<img width="306" alt="Screen Shot 2021-04-26 at 3 02 30 PM" src="https://user-images.githubusercontent.com/24382050/116042153-abaaae80-a6a0-11eb-9f40-f1b9f49fb4f3.png">
<img width="311" alt="image" src="https://user-images.githubusercontent.com/13028267/116976929-5d2a8f00-acf4-11eb-9f10-6fc2f49ee232.png">

Commit:
Initial BarChart

Commit:
Merge remote-tracking branch 'origin/feature/LineChart' into feature/BarChart

Commit:
optimize imports

Commit:
resolved private import issue

Commit:
add spinner to select 1 or more data sources

Commit:
add stacked and normal for data size of 2

Commit:
preview

Commit:
preview and more added info for barchart

Commit:
preview and more added info for barchart

Commit:
fix space

Commit:
fix proper spacing for 1 datatype

Commit:
add params for BarChart

Commit:
Merge remote-tracking branch 'origin/develop' into feature/BarChart

Commit:
added instrumented Tests

Commit:
added all vertical Fragment samples

rickimaru18 committed:
Merge branch 'develop' into feature/BarChart

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt:
```diff
@@ -1,18 +1,277 @@
+                size = Size(width = 48f, height = size.height - point.y),
+                color = barSeries.color,
+                style = Stroke(barSeries.width.toFloat()),
+//              brush = Brush.linearGradient(colors = listOf(barSeries.areaColor!!))
```
remove if not used

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rickimaru18 committed:
Refactor.

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt:
```diff
@@ -1,18 +1,277 @@
+                        .padding(
+                                start = (mMaxVal.toString().length * 8).dp,
+                                top = 10.dp,
+                                end = 10.dp,
```
create a diff function for this

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt:
```diff
@@ -1,18 +1,277 @@
+private fun DrawScope.drawGrids(count: Int, color: Color) {
+    val startPoint = origin()
+//    val pathEffect = PathEffect.dashPathEffect(intervals = floatArrayOf(10f, 10f))
+
```
remove if not used

rexendy replied:
remove all commented codes if not used

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt:
```diff
@@ -1,18 +1,277 @@
+            it.nativeCanvas.drawText(
+                    (valuePerGrid * (i + 1)).toInt().toString(),
+                    valOffset.x,
+                    valOffset.y,
```
if possible make a function for this

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/model/BarSeries.kt:
```diff
@@ -0,0 +1,39 @@
+ * LineChart data.
+ */
+data class BarSeries<T>(
+        /**
```
apply proper spacing
/** pls align it to its closing tag

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on sample/src/main/res/layout/fragment_bar.xml:
```diff
@@ -6,17 +6,22 @@
-    <TextView
-        android:id="@+id/text_dashboard"
+    <androidx.compose.ui.platform.ComposeView
+        android:id="@+id/chartBar"
```
pls use snake case convention for xml id naming

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on sample/src/main/java/com/kwarchart/sample/ui/bar/BarFragment.kt:
```diff
@@ -4,21 +4,26 @@ import android.os.Bundle
-class BarFragment : Fragment() {
+
+
+
```
remove extra line breaks

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt:
```diff
@@ -166,15 +197,20 @@ private fun DrawScope.drawGrids(count: Int, color: Color) {
+//                start = Offset(i * mHGap, 0f),
+//                end = Offset(i * mHGap, size.height),
+////            pathEffect = pathEffect
+//        )
```
remove if not used

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy commented on kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt:
```diff
@@ -288,15 +288,36 @@ fun BarChartPreview() {
+
+            ///TODO
+//            type = HORIZONTAL,
+//            type = HORIZONTAL_STACKED,
```
remove if not used

rickimaru18 replied:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.

rexendy reviewed (changes requested):
Hello Ty

- pls check spacing indentation and line breaks
coordindate it with Rick to have same format
- pls add all static text to strings.xml
- pls remove all commented codes that are not used
- if possible, make your functions not too long ( optimize only for those
functions needed)

Nice test. thank you

rickimaru18 commented:
@totzk9 @rexendy
Bugs found:
1. Stack display incorrect. Should stack to Y-axis, not Z-axis.
2. Incorrect display when 3 or more BarSeries .

I'll just fix this when I implement HORIZONTAL BarChart.

rickimaru18 commented:
To be continued in horizontal_bar_chart branch.

rickimaru18 committed:
Refactor.

rickimaru18 committed:
Fix VERTICAL BarChart

# In BarChart
- Support more than 2 BarSeries.
- Fix VERTICAL_STACKED type. Will not stack in Y-axis instead of Z-axis.

rickimaru18 commented:
> Hello Ty
> 
>     * pls check spacing indentation and line breaks
>       coordindate it with Rick to have same format
> 
>     * pls add all static text to strings.xml
> 
>     * pls remove all commented codes that are not used
> 
>     * if possible, make your functions not too long ( optimize only for those
>       functions needed)
> 
> 
> Nice test. thank you

@rexendy 
CC: @totzk9 

Please see 1717eb9 and 37d0304 :bow:

rexendy commented:
> @totzk9 @rexendy
> Bugs found:
> 
> 1. Stack display incorrect. Should stack to Y-axis, not Z-axis.
> 2. Incorrect display when 3 or more BarSeries .
> 
> I'll just fix this when I implement HORIZONTAL BarChart.

nice catch @rickimaru18

rexendy commented:
@rickimaru18 pls merge it to develop branch.

Thank you
